### ISSUE TITLE
Why SequenceExtensions is EditorBrowsableState.Never?

### ISSUE THREAD

**Issue by kostrse** [moq/moq4#1022]
I was looking for extension methods like **ReturnsAsync** and **ThrowsAsync** for **ISetupSequentialResult** and found that the SequenceExtensions class is marked as `EditorBrowsableState.Never`?

What is the reasoning behind it?
It's not marked as obsolete and I didn't find any existing issues or pull requests discussing it.

**Comment by stakx** [comment]
This is a pattern that you see often in .NET code. The container class for extension methods is often irrelevant, we don't want to pollute IntelliSense with an irrelevant entry.

It shouldn't matter whether a method is regular or an extension, they will still show up in the extended type's IntelliSense method list and that's where you'll find them.

If we really wanted to make extension method containers browsable (and, for the above reasons, that isn't necessary), we'd have a different problem to solve. Moq extension methods are spread over various classes evwn where they extend the same type; that's a mistake. Fixing it, i.e. merging some of the extension method classes, would be a breaking change, and for that reason likely won't happen in the 4.x version line.

**Comment by kostrse** [comment]
So, the intention wasn't to hide the methods themselves but to hide the container class only?

I'm using Rider and it wasn't showing any of these methods with autocomplete, so it took me to look into the sources and I discovered the `EditorBrowsableState.Never` attribute, I got confused and started to think that these methods were intentionally hidden for some reason.

After your response, I checked in Visual Studio and it shows the methods just fine. After explicitly typing them in Rider, rider also discovered them later and shows in autocomplete. Looks like like it's just a bug in Rider or corrupted cache.
